workflow-core/src/commonMain/kotlin/com/squareup/workflow1/WorkflowAction.kt

```diff
@@ -31,6 +31,9 @@ import kotlin.jvm.JvmOverloads
  */
 public abstract class WorkflowAction<in PropsT, StateT, out OutputT> {
 
+  // Debugging name, will be returned by `toString()`
+  public open val name: String = CommonKClassTypeNamer.uniqueName(this::class)
```

Comment:
we could, but we don't have control over that. if someone makes a massive `data class MyAction: WorkflowAction` with a tonne of members (as we have seen) then `toString()` will still print all those out.

`final toString()` prevents that and make sure that `WorkflowAction` logging is always based on the `name` which can only be so big (in reality)

Reply:
Right, but I'm saying the real problem is that we're using `toString()` in logging at all, which we do because we had no alternative. But now `name` exists and we can log that — avoiding any call to `toString()` in production code as nature intended. That seems like a complete solution, I don't see why we need to mess with `toString()` as well.

Reply:
> Right, but I'm saying the real problem is that we're using toString() in logging at all, which we do because we had no alternative. But now name exists and we can log that — avoiding any call to toString() in production code as nature intended. That seems like a complete solution, I don't see why we need to mess with toString() as well.


Ahhh IC. that makes a tonne of sense